operator/pkg/reconciliation/reconcile_racks.go

```diff
@@ -825,6 +845,36 @@ func (rc *ReconciliationContext) CreateUsers() result.ReconcileResult {
 	return result.Continue()
 }
 
+func (rc *ReconciliationContext) ConfigureSystemAuthReplication() result.ReconcileResult {
+	dc := rc.Datacenter
+
+	if dc.Spec.Stopped {
+		rc.ReqLogger.Info("cluster is stopped, skipping ConfigureSystemAuthReplication")
+		return result.Continue()
+	}
+
+	pod := rc.dcPods[0]
+
+	replication := []httphelper.ReplicationSetting{
+		{
+			Datacenter: dc.Name,
+			ReplicationFactor: getSystemAuthRF(dc),
+		},
+	}
+
+	err := rc.NodeMgmtClient.CallAlterKeyspaceEndpoint(pod, "system_auth", replication)
+
+	if err != nil {
+		return result.Error(err)
+	}
+
+	return result.Continue()
+}
+
+func getSystemAuthRF(dc *api.CassandraDatacenter) int {
+	return int(math.Min(float64(3), float64(dc.Spec.Size)))
```

Comment:
I wonder if we should use the number of racks, instead of the number of nodes, in this `math.Min()` call. In the `isClusterHealthy()` function, we seem to use the rack count as just such a proxy for the RF:

```
func (rc *ReconciliationContext) isClusterHealthy() bool {
	pods := FilterPodListByCassNodeState(rc.clusterPods, stateStarted)

	numRacks := len(rc.Datacenter.GetRacks())
	for _, pod := range pods {
		err := rc.NodeMgmtClient.CallProbeClusterEndpoint(pod, "LOCAL_QUORUM", numRacks)
		if err != nil {
			return false
		}
	}

	return true
}
```

So maybe `return int(math.Min(float64(3), float64(dc..GetRacks())))` instead?

Reply:
@johntrimble what about the scenario though in which we have just a single rack?

Reply:
What I'm observing in the code is that we use the number of racks to indicate the intended RF, as in the example above. Is using the number of racks not a suitable proxy for the intended RF? My expectation would be that the only case in which we would not have 3 racks would be the case in which a user is creating a cluster of only 1 node. Is there a reason a user would want to create a cluster of 3+ nodes with only a single rack? 

It's possible the code example I gave above is what needs to change, but I'd like for us to do this in a consistent manner.

Reply:
> Is using the number of racks not a suitable proxy for the intended RF?

Having RF = num_racks is definitely a good practice.

> My expectation would be that the only case in which we would not have 3 racks would be the case in which a user is creating a cluster of only 1 node. Is there a reason a user would want to create a cluster of 3+ nodes with only a single rack?

My understanding is that the operator allows you to specify both the number of nodes and the number of racks. I know that the operator will try to create balanced racks, but the key thing for me here is that there is nothing preventing me from creating a C* cluster with just one rack. If I am only running in a single zone, then a single rack is fine with respect to C* topology.

If we determine the RF of system_auth by the number of racks, then we can wind up with RF = 1 for system_auth even if there are multiple nodes. That would be bad, and not just for Reaper.